backend/danswer/dynamic_configs/store.py

```diff
@@ -56,12 +60,19 @@ def delete(self, key: str) -> None:
 class PostgresBackedDynamicConfigStore(DynamicConfigStore):
     @contextmanager
     def get_session(self) -> Iterator[Session]:
-        factory = get_session_factory()
-        session: Session = factory()
-        try:
+        engine = get_sqlalchemy_engine()
+        with Session(engine, expire_on_commit=False) as session:
+            if MULTI_TENANT:
+                tenant_id = current_tenant_id.get()
+                if tenant_id == "public":
+                    raise HTTPException(
+                        status_code=401, detail="User must authenticate"
+                    )
+                if not is_valid_schema_name(tenant_id):
+                    raise HTTPException(status_code=400, detail="Invalid tenant ID")
+                # Set the search_path to the tenant's schema
+                session.execute(text(f'SET search_path = "{tenant_id}"'))
             yield session
-        finally:
```

Comment:
Why do we remove this?

Reply:
Remember why I removed this - `with session` should handle this robustly for us